### PR TITLE
SAK-50642 Content fix folder permission editing

### PR DESCRIPTION
**Pull request by ottenhoff**
I haven't tested this much. Just first working draft.

Maybe it would be nice to look for permissions in the parent folder to display (if no custom realm exists for the site)?

### CODE REVIEW

File: webcomponents/tool/src/main/java/org/sakaiproject/webcomponents/permissions/PermissionsEntityProvider.java [hunk 1]
```diff
@@ -161,8 +173,22 @@ public String handleSet(EntityReference entityRef, Map<String, Object> params) {
 
         String reference = (String) params.get("ref");
 
+        AuthzGroup authzGroup;
+        try {
+            authzGroup = authzGroupService.getAuthzGroup(reference);
+        } catch (GroupNotDefinedException e) {
+            try {
+                authzGroup = authzGroupService.addAuthzGroup(reference);
```
Comment (adrianfish) [inline webcomponents/tool/src/main/java/org/sakaiproject/webcomponents/permissions/PermissionsEntityProvider.java:181]: This doesn't seem a great way to do things, imo. Surely if content hosting relies on realms for its folder permissions, it should create them as the folders are created?

Comment (ottenhoff) [reply webcomponents/tool/src/main/java/org/sakaiproject/webcomponents/permissions/PermissionsEntityProvider.java:181]: custom permissions per folder is a 1% use case .... so if you created a new realm per folder, you would have hundreds of thousands of new realms that are unnecessary. and then you start introducing new edge case bugs: e.g., what happens when you adjust Resources permissions to remove content.read but all of your folders are still readable?

Comment (adrianfish) [reply webcomponents/tool/src/main/java/org/sakaiproject/webcomponents/permissions/PermissionsEntityProvider.java:181]: Should the reference be checked that it is actually under the current site, if you know what I mean?

Comment (ottenhoff) [reply webcomponents/tool/src/main/java/org/sakaiproject/webcomponents/permissions/PermissionsEntityProvider.java:181]: yes i think prob a good addition. perms are checked later on and obviously logged .... but sure makes sense

Comment (ottenhoff) [reply webcomponents/tool/src/main/java/org/sakaiproject/webcomponents/permissions/PermissionsEntityProvider.java:185]: updated .....

**Comment by ottenhoff** [comment]
@adrianfish are you okay with this now?

**Pull request merged**